server/src/test/java/org/elasticsearch/health/node/FetchHealthInfoCacheActionTests.java

```diff
@@ -146,16 +151,19 @@ public void testResponseSerialization() {
     }
 
     private FetchHealthInfoCacheAction.Response mutateResponse(FetchHealthInfoCacheAction.Response originalResponse) {
-        Map<String, DiskHealthInfo> diskHealthInfoMap = originalResponse.getHealthInfo().diskInfoByNode();
-        Map<String, DiskHealthInfo> diskHealthInfoMapCopy = new HashMap<>(diskHealthInfoMap);
-        diskHealthInfoMapCopy.put(
-            randomAlphaOfLength(10),
-            new DiskHealthInfo(randomFrom(HealthStatus.values()), randomFrom(DiskHealthInfo.Cause.values()))
-        );
         return new FetchHealthInfoCacheAction.Response(
             new HealthInfo(
-                diskHealthInfoMapCopy,
-                randomValueOtherThan(originalResponse.getHealthInfo().dslHealthInfo(), HealthInfoTests::randomDslHealthInfo)
+                mutateMap(
+                    originalResponse.getHealthInfo().diskInfoByNode(),
+                    () -> randomAlphaOfLength(10),
+                    HealthInfoTests::randomDiskHealthInfo
+                ),
+                randomValueOtherThan(originalResponse.getHealthInfo().dslHealthInfo(), HealthInfoTests::randomDslHealthInfo),
+                mutateMap(
+                    originalResponse.getHealthInfo().repositoriesInfoByNode(),
+                    () -> randomAlphaOfLength(10),
+                    HealthInfoTests::randomRepoHealthInfo
+                )
             )
         );
```

Comment:
Now that we have more values we should change this mutate function to only change one parameter at a time. I see that was not the case before but it's never too late :p